dask_expr/_shuffle.py

```diff
@@ -804,6 +819,10 @@ class SetIndex(BaseSetIndexSortValues):
     }
     _filter_passthrough = True
 
+    @functools.cached_property
+    def unique_partition_mapping_columns(self):
+        return {tuple(self.other.columns)}
```

Comment:
Coming back to my argument about how this condition can be generated from different functions... I am not sure if this is actually correct.

IIUC set_index works by calculating the divisions and the sorting the partitions accordingly. This will create a different data distribution than if we were to shuffle on a couple of columns

Reply:
Yep that's correct, set-index works differently (and is not included in this implementation fwiw).

We probably need a more general purpose propagation implementation. All methods that propagate the current mapping will also propagate the set_index or parquet mappings, but that can be a follow up when we add those